### ISSUE TITLE
AlgorithmID is overwritten

### ISSUE THREAD

**Issue by Doggie52** [QuantConnect/Lean#3011]
#### Expected Behavior
The `AlgorithmId` set in `Initialize()` using `SetAlgorithmId()` is kept constant throughout the life of the algorithm.

#### Actual Behavior
`AlgorithmId` is overwritten at https://github.com/QuantConnect/Lean/blob/642437d030b161e11bf979ea2889bd9bc9b5499c/Engine/Engine.cs#L271.

#### Potential Solution
Perhaps the `job.AlgorithmId` should try to pull its value from the algorithm first?

#### Reproducing the Problem
In `Initialize()` execute `SetAlgorithmId( "test" );`. Somewhere outside of this function, debug and inspect the value of `AlgorithmId`.

#### System Information
Windows x64.

#### Related Issue
#2472

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

**Comment by Martin-Molinero** [comment]
Hey @Doggie52!
Believe `SetAlgorithmId()` isn't intended to be called by the algorithm itself. Recently we merged PR https://github.com/QuantConnect/Lean/pull/4156 where we added a couple of configuration options and cmdline arguments `"algorithm-id"` & `"results-destination-folder"`, which I think should help solve the underlying issue and avoid calling `SetAlgorithmId()`. Let us know if this solves the issue and we can close it
